### PR TITLE
Mention javascript #22274

### LINKED ISSUE #22274: No mention of JavaScript on this page, when it's one the only way to make it do something (other than using it inside of a form)
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button

### What specific section or headline is this issue about?

First explanatory section

### What information was incorrect, unhelpful, or incomplete?

I think it should have a mention of JavaScript.

### What did you expect to see?

Having a mention of JavaScript, like rewriting the current sentence:

> Once activated, it then performs a programmable action, such as submitting a [form](https://developer.mozilla.org/en-US/docs/Learn/Forms) or opening a dialog.

To something like
 Once activated, it submits a form [form](https://developer.mozilla.org/en-US/docs/Learn/Forms) or performs a programmable action using [JavaScript](https://developer.mozilla.org/en-US/docs/Learn/JavaScript), such as opening a dialog.

### Do you have any supporting links, references, or citations?

No

### Do you have anything more you want to share?

No

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/html/element/button`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/html/element/button/index.md
* Last commit: https://github.com/mdn/content/commit/7594b50698a76ce79209b159835e615052915723
* Document last modified: 2022-10-31T06:32:10.000Z

</details>

### PR DESCRIPTION
**Pull request by kylewcode**
### Description

Added mention of Javascript in first description paragraph using a link to the learning area for Javascript.

### Motivation

Help new users to connect programmability on the web with Javascript. Contribute to open source, practice pull requests, get faster at comprehending issues.

### Related issues and pull requests
Fixes #22274

### CODE REVIEW

**Comment by MTindev** [comment]
Hi @kylewcode,

Two comments about the PR : 

1) Above all, "Javascript" must be written "JavaScript".

2) I find the new proposal less relevant than the current one or @nathnolt's original proposal in #22274.

> Once activated, it performs a programmable action using Javascript, such as submitting a form or opening a dialog.

This suggests that a programmable action is *required* in JavaScript.

However, a button, depending on its configuration (the value of its attribute type in particular) allows to perform native actions without JavaScript: the submission of the form or the reset of the fields.

Beyond these native actions, it is indeed relevant to specify that a programmable action in JavaScript is possible (and it is often the case), but this is complementary to native behavior.

**Comment by estelle** [comment]
> Beyond these native actions, it is indeed relevant to specify that a programmable action in JavaScript is possible (and it is often the case), but this is complementary to native behavior.

I agree with @MTindev's assessment. We can definitely include JS in this page, but it should not be in the first paragraph as the first paragraph is the summary of the element used in many other places (debuggers, vs code, extensions, etc., all use the first paragraph from MDN as a summary in their apps). The button can do things without any JS; so JS is not a requirement, so not main summary worthy.

Indeed, the events that are triggered should be included in this page; I just don't think this suggestion is the right way to include it.

File: files/en-us/web/html/element/button/index.md [hunk 1]
```diff
@@ -185,6 +185,8 @@ A submit button with the attribute `formaction` set, but without an associated f
 
 If your buttons are not for submitting form data to a server, be sure to set their `type` attribute to `button`. Otherwise they will try to submit form data and to load the (nonexistent) response, possibly destroying the current state of the document.
 
+Although not required, an activated button can also perform a programmable action using [JavaScript](/en-US/docs/Learn/JavaScript), such as removing an item from a list.
```
Comment (estelle) [inline files/en-us/web/html/element/button/index.md:188]: How about:

```suggestion
While `<button type="button">` has no default behavior, event handlers can be scripted to trigger behaviors.  An activated button can perform programmable actions using [JavaScript](/en-US/docs/Learn/JavaScript), such as removing an item from a list.
```

**Comment by MTindev** [comment]
Looks great. For me the  suggestions of reviewed proposal reflects well the expectation 👍

File: files/en-us/web/html/element/button/index.md [hunk 2]
```diff
@@ -185,6 +185,8 @@ A submit button with the attribute `formaction` set, but without an associated f
 
 If your buttons are not for submitting form data to a server, be sure to set their `type` attribute to `button`. Otherwise they will try to submit form data and to load the (nonexistent) response, possibly destroying the current state of the document.
 
+While `<button type="button">` has no default behavior, event handlers can be scripted to trigger behaviors.  An activated button can perform programmable actions using [JavaScript](/en-US/docs/Learn/JavaScript), such as removing an item from a list.
```
Comment (estelle) [inline files/en-us/web/html/element/button/index.md:188]:
```suggestion
While `<button type="button">` has no default behavior, event handlers can be scripted to trigger behaviors. An activated button can perform programmable actions using [JavaScript](/en-US/docs/Learn/JavaScript), such as removing an item from a list.
```

**Review by estelle — APPROVED** [review]
thank you

**Comment by kylewcode** [comment]
You're welcome!

**Pull request merged**